nemo_skills/inference/generate.py

```diff
@@ -422,60 +397,8 @@ def fill_prompt(self, data_point, data):
-        generation_params = {
-            "prompts": [self.fill_prompt(dp, data) for dp in data_points],
-            "stop_phrases": combine_stop_phrases(
-                self.prompt.stop_phrases if self.prompt is not None else None, self.extra_stop_phrases
-            ),
-            **asdict(self.cfg.inference),
-            **self.extra_generate_params,
-        }
-
-        if self.cfg.code_execution:
-            if self.cfg.override_max_code_executions and self.cfg.total_code_executions_in_prompt is not None:
-                max_code_executions_values = [dp['total_code_executions'] for dp in data_points]
-                generation_params['max_code_executions'] = max_code_executions_values
 
-        generate_method = self.llm.generate_async if is_async else self.llm.generate
-
-        return generate_method(**generation_params)
-
-    # TODO: rewrite mtbench to have turns separated in data file and remove this method
-    def llm_generate_multi_turn(self, data_points, data):
-        # TODO: this will not be efficient if different elements have different number of turns
-        # (effective batch size gets smaller). Need to rewrite it to ensure batch size is filled
-        # no matter the turns. Also even the below implementation can probably be simplified
-        turn_data_points = deepcopy(data_points)
-        dp_indices = list(range(len(turn_data_points)))
-        cur_turn = 1
-        outputs = [{"generation": []} for _ in range(len(data_points))]
-        while dp_indices:
-            # updating the turns to only have data up-to the current turn
-            # and adding any generated assistant messages
-            for dp_index in dp_indices:
-                turn_data_points[dp_index][self.cfg.multi_turn_key] = data_points[dp_index][self.cfg.multi_turn_key][
-                    :cur_turn
-                ]
-                for turn_idx in range(cur_turn - 1):
-                    turn_data_points[dp_index][self.cfg.multi_turn_key][turn_idx]['assistant'] = outputs[dp_index][
-                        "generation"
-                    ][turn_idx]
-            # getting a new set of generations
-            turn_outputs = self.llm_generate([turn_data_points[dp_index] for dp_index in dp_indices], data)
-            # adding assistant answers to the generations
-            for pos_index, dp_index in enumerate(dp_indices):
-                outputs[dp_index]["generation"].append(turn_outputs[pos_index]["generation"])
-
-            # removing any indices that got through all turns
-            dp_indices = []
-            for dp_index, (output, dp) in enumerate(zip(outputs, data_points)):
-                if len(output["generation"]) < len(dp[self.cfg.multi_turn_key]):
-                    dp_indices.append(dp_index)
-            cur_turn += 1
-        return outputs
-
-    def dump_outputs(self, outputs, data_points, fout):
+    def _dump_outputs(self, outputs, data_points, fout):
```

Comment:
I feel the current naming with and without _ is a bit inconsistent. Typically, underscore is used for public methods that can be called outside of the class. In that case, I'd say everything should start with underscore here except "generate" as no other function can really be called in isolation. So either we only have generate without underscore or we have everything without underscore (which was the original naming where we basically just don't differentiate between private and public). So we should be consistent in naming here ideally